Add check to keep releases.json in sync with the firmware binaries

peter-sanderson opened the pull request:
This PR adds check to validate presence of the binaries as declared in the `releases.json`

If some binary is missing:
![image](https://github.com/user-attachments/assets/95363bb5-0df8-4ad8-a876-328cfffff541)

If there is binary but is missing from `releases.json`
![image](https://github.com/user-attachments/assets/fc58defe-7703-4897-975d-95fdcaa8d095)



It also adds a `shellcheck`

peter-sanderson commented on ci/s3sync.sh:
```diff
@@ -6,11 +6,9 @@
 
 DIRS="bootloader bridge firmware legal registry udev suite connect security transparency misc"
 BUCKET=data.trezor.io
-ROLLBACK=rollback-data.trezor.io
```
Unused (`shellcheck` told me)

peter-sanderson commented on firmware/t3b1/releases.json:
```diff
@@ -0,0 +1 @@
+[]
```
Is this safe to add empty? Its just for consistency.

komret replied:
Should be safe.

peter-sanderson commented on ci/s3sync.sh:
It may be worth to unify `ci` and `scripts` folders? @vdovhanych WDYT?

vdovhanych replied:
yeah, we can move the release script into the scripts folder, but it's not necessary, tbh.

komret commented on scripts/run-releases-json-for-all-devices.sh:
Could this be included in the `check-firmware-presence-in-releases-json.sh` file, calling `compare_files ` in the loop?

peter-sanderson replied:
I liked I can just run it one-by one when using it locally. But this is ok as well WDYT?

komret replied:
OK

komret reviewed (dismissed):
The shellcheck addition could have been a separate commit.

vdovhanych commented on .github/workflows/check-shell-validation.yml:
```diff
@@ -0,0 +1,19 @@
+name: "[Check]: Shell validation"
```
remove `:`

vdovhanych reviewed (changes requested):
Adding shellcheck validation should be a separate commit and please fix one small nitpick.

peter-sanderson committed:
chore(ci): Add sellcheck workflow

peter-sanderson committed:
chore(ci): Add check to keep releases.json in sync with the firmware binaries

peter-sanderson commented on scripts/check-firmware-presence-in-releases-json.sh:
```diff
@@ -0,0 +1,90 @@
+trezor-t1b1-1.3.2.bin
+trezor-t1b1-1.3.3.bin
+trezor-t1b1-1.3.4.bin
+trezor-t1b1-1.3.5.bin
```
@prusnak We are not having those old FW binary files. Is it ok to just ignore them from this check?

We have them in `releases.json` so this check expects them in the directory, but they are not here.

szymonlesisz replied:
i think we should remove it from `releases.json` as they are not supported (<1.9.0)

peter-sanderson replied:
OK, did it here: https://github.com/trezor/data/pull/90/commits/b8000706a8c58bb2518052b3456eca5b87c2e8a4

peter-sanderson committed:
chore(ci): Remove unsupported super-old firmware versions for Trezor One

peter-sanderson committed:
fix: rework the algorithm so it takes the  from releases.json into an account, this allows a cross-directory links

fix: binary missing in 2 directory + fix inconsistend url in releases.json

fix: rework the algorithm so it takes the  from releases.json into an account, this allows a cross-directory links

komret reviewed (dismissed):
I'd rather see this integrated with `check_releases.py`, but now that it is done like this, let's use it. @vdovhanych, can you approve?

vdovhanych reviewed (dismissed):
i guess its fine like this for now

peter-sanderson commented on firmware/2/releases.json:
```diff
@@ -8,10 +8,10 @@
         "fingerprint": "d3af84a212d32785449ca6575e3cf2a641920b353a82dec9f059083ea5d4b149",
         "changelog": "* Fixed displaying of a progress indicator for the formatting operation.\n* Improve precision of PIN validation countdown.\n* Solana: Improved support for AToken Create operation.",
-        "url_bitcoinonly": "data/firmware/t2t1/trezor-t2t1-2.8.1-bitcoinonly.bin",
+        "url_bitcoinonly": "data/firmware/2/trezor-2.8.1-bitcoinonly.bin",
```
@matejcik :pray: is this ok? Or did you done this deliberately?

Lemonexe reviewed (approved):
LGTM :rocket:
